src/remote.c

```diff
@@ -278,19 +280,19 @@ static int create_internal(git_remote **out, git_repository *repo, const char *n
 	else
 		remote->download_tags = GIT_REMOTE_DOWNLOAD_TAGS_AUTO;
 
-
-	git_buf_dispose(&var);
-
+	if (txn)
+		git_transaction_commit(txn);
```

Comment:
Sorry to not have catched this earlier, but you need to do error checking here. Otherwise we might end up silently not writing the config changes.